packages/@aws-cdk/cdk/lib/encoding.ts

```diff
@@ -200,9 +200,79 @@ export function containsListTokenElement(xs: any[]) {
 export function unresolved(obj: any): boolean {
   if (typeof(obj) === 'string') {
     return TokenString.forStringToken(obj).test();
+  } else if (typeof obj === 'number') {
+    return extractTokenDouble(obj) !== undefined;
   } else if (Array.isArray(obj) && obj.length === 1) {
     return typeof(obj[0]) === 'string' && TokenString.forListToken(obj[0]).test();
   } else {
     return obj && typeof(obj[RESOLVE_METHOD]) === 'function';
   }
-}
\ No newline at end of file
+}
+
+/**
+ * Bit pattern in the top 16 bits of a double to indicate a Token
+ *
+ * An IEEE double in LE memory order looks like this (grouped
+ * into octets, then grouped into 32-bit words):
+ *
+ * mmmmmmm.mmmmmmm.mmmmmmm.mmmmmmm | mmmmmmm.mmmmmmm.EEEEEmm.sEEEEEE
+ *
+ * - m: mantissa (52 bits)
+ * - E: exponent (11 bits)
+ * - s: sign (1 bit)
+ *
+ * We put the following marker into the top 16 bits (exponent and sign), and
+ * use the mantissa part to encode the token index. To save some bit twiddling
+ * we use all top 16 bits for the tag. That loses us 2 mantissa bits to store
+ * information in but we still have 50, which is going to be plenty for any
+ * number of tokens to be created during the lifetime of any CDK application.
+ *
+ * Can't have all bits set because that makes a NaN, so unset the least
+ * significant exponent bit.
+ *
+ * Currently not supporting BE architectures.
+ */
+// tslint:disable-next-line:no-bitwise
+const DOUBLE_TOKEN_MARKER_BITS = 0xFBFF << 16;
+
+/**
+ * Return a special Double value that encodes the given integer
+ */
```

Comment:
Mention in the docstring that this is used to encode an _index_ to the token map and therefore it's okay that we don't support negative numbers. Also, mention the limit on the number of number tokens we can support.